extensions/spark/kyuubi-spark-connector-tpch/src/test/scala/org/apache/kyuubi/spark/connector/tpch/TPCHQuerySuite.scala

```diff
@@ -0,0 +1,149 @@
+    "q6" -> "`revenue` DOUBLE",
+    "q12" -> "`l_shipmode` STRING,`high_line_count` BIGINT,`low_line_count` BIGINT")
+
+  val queryToSchema_3_2: Map[String, String] = queryToUnchangedSchema ++ Map(
+    "q4" -> "`o_orderpriority` STRING,`order_count` BIGINT NOT NULL",
+    "q22" -> "`cntrycode` STRING,`numcust` BIGINT NOT NULL,`totacctbal` DOUBLE",
+    "q21" -> "`s_name` STRING,`numwait` BIGINT NOT NULL",
+    "q13" -> "`c_count` BIGINT NOT NULL,`custdist` BIGINT NOT NULL",
+    "q1" -> ("`l_returnflag` STRING,`l_linestatus` STRING,`sum_qty` DOUBLE," +
+      "`sum_base_price` DOUBLE,`sum_disc_price` DOUBLE,`sum_charge` DOUBLE," +
+      "`avg_qty` DOUBLE,`avg_price` DOUBLE,`avg_disc` DOUBLE,`count_order` BIGINT NOT NULL"),
+    "q16" -> "`p_brand` STRING,`p_type` STRING,`p_size` INT,`supplier_cnt` BIGINT NOT NULL")
+
+  val queryToSchema: Map[String, String] = queryToUnchangedSchema ++ Map(
+    "q4" -> "`o_orderpriority` STRING,`order_count` BIGINT",
+    "q22" -> "`cntrycode` STRING,`numcust` BIGINT,`totacctbal` DOUBLE",
+    "q21" -> "`s_name` STRING,`numwait` BIGINT",
+    "q13" -> "`c_count` BIGINT,`custdist` BIGINT",
+    "q1" -> ("`l_returnflag` STRING,`l_linestatus` STRING,`sum_qty` DOUBLE," +
+      "`sum_base_price` DOUBLE,`sum_disc_price` DOUBLE,`sum_charge` DOUBLE," +
+      "`avg_qty` DOUBLE,`avg_price` DOUBLE,`avg_disc` DOUBLE,`count_order` BIGINT"),
+    "q16" -> "`p_brand` STRING,`p_type` STRING,`p_size` INT,`supplier_cnt` BIGINT")
+
+  test("run query on tiny") {
+    val viewSuffix = "view";
+    val sparkConf = new SparkConf().setMaster("local[*]")
+      .set("spark.ui.enabled", "false")
+      .set("spark.sql.catalogImplementation", "in-memory")
+      .set("spark.sql.catalog.tpch", classOf[TPCHCatalog].getName)
+    withSparkSession(SparkSession.builder.config(sparkConf).getOrCreate()) { spark =>
+      spark.sql("USE tpch.tiny")
+      queries.map { queryName =>
+        val in = getClass.getClassLoader.getResourceAsStream(
+          s"tpch/$queryName.sql")
+        val queryContent: String = Source.fromInputStream(in)(Codec.UTF8).mkString
+        in.close()
+        queryName -> queryContent
+      }.foreach { case (name, sql) =>
+        try {
+          val result = spark.sql(sql).collect()
+          val schema = spark.sql(sql).schema
+          spark.createDataFrame(result.toList.asJava, schema).createTempView(s"$name$viewSuffix")
+          val sumHashResult =
+            spark.sql(s"select sum(hash(*)) from $name$viewSuffix").collect().head.get(0)
+          // scalastyle:off
+          println(s"name=$name,schema=${schema.toDDL},sumHashResult=$sumHashResult")
+          // scalastyle:on
+          assert(sumHashResult == queryToSumHash(name))
+          if (SparkUtils.isSparkVersionEqualTo("3.3")) {
+            // The toDDL method of StructType, the return value in spark3.3
+            // is different from that in spark3.2
+            // in spark3.2 "`a` INT"
+            // in spark3.3 "a INT"
```

Comment:
It's fragile to use String to compare schema, would you please try to compare `StructType` directly?

Reply:
anther option is maitain the output schema in seprate files like `OutputSchemaTPCDSSuite` does.